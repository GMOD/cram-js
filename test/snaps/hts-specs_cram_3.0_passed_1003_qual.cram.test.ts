import { testFile } from '../testUtil'
testFile('hts-specs/cram/3.0/passed/1003_qual.cram')
