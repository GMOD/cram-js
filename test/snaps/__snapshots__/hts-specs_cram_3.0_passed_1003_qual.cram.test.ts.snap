// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`can dump the whole hts-specs/cram/3.0/passed/1003_qual.cram without error 1`] = `
[
  {
    "fileId": "_.cram",
    "magic": "CRAM",
    "majorVersion": 3,
    "minorVersion": 0,
  },
  {
    "containerHeader": {
      "_endPosition": 45,
      "_size": 19,
      "alignmentSpan": 0,
      "crc32": 3348343310,
      "landmarks": [
        0,
        163,
      ],
      "length": 253,
      "numBases": 0,
      "numBlocks": 2,
      "numLandmarks": 2,
      "numRecords": 0,
      "recordCounter": 0,
      "refSeqId": 0,
      "refSeqStart": 0,
    },
    "data": [
      {
        "_endPosition": 45,
        "_size": 19,
        "alignmentSpan": 0,
        "crc32": 3348343310,
        "landmarks": [
          0,
          163,
        ],
        "length": 253,
        "numBases": 0,
        "numBlocks": 2,
        "numLandmarks": 2,
        "numRecords": 0,
        "recordCounter": 0,
        "refSeqId": 0,
        "refSeqStart": 0,
      },
      {
        "samHeader": [
          {
            "data": [
              {
                "tag": "SN",
                "value": "CHROMOSOME_I",
              },
              {
                "tag": "LN",
                "value": "1009800",
              },
              {
                "tag": "M5",
                "value": "8ede36131e0dbf3417807e48f77f3ebd",
              },
              {
                "tag": "UR",
                "value": "/nfs/users/nfs_j/jkb/work/samtools_master/hts-specs/test/cram/passed/../ce.fa",
              },
            ],
            "tag": "SQ",
          },
        ],
      },
      {
        "samHeader": [
          {
            "data": [
              {
                "tag": "SN",
                "value": "CHROMOSOME_I",
              },
              {
                "tag": "LN",
                "value": "1009800",
              },
              {
                "tag": "M5",
                "value": "8ede36131e0dbf3417807e48f77f3ebd",
              },
              {
                "tag": "UR",
                "value": "/nfs/users/nfs_j/jkb/work/samtools_master/hts-specs/test/cram/passed/../ce.fa",
              },
            ],
            "tag": "SQ",
          },
        ],
      },
    ],
  },
  {
    "compressionScheme": {
      "APdelta": true,
      "dataSeriesEncoding": {
        "AP": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 17,
          },
          "parametersBytes": 1,
        },
        "BA": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 30,
          },
          "parametersBytes": 1,
        },
        "BB": {
          "codecId": 4,
          "parameters": {
            "lengthsEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 42,
              },
              "parametersBytes": 1,
            },
            "valuesEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 37,
              },
              "parametersBytes": 1,
            },
          },
          "parametersBytes": 6,
        },
        "BF": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 15,
          },
          "parametersBytes": 1,
        },
        "BS": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 31,
          },
          "parametersBytes": 1,
        },
        "CF": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 16,
          },
          "parametersBytes": 1,
        },
        "FC": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 27,
          },
          "parametersBytes": 1,
        },
        "FN": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 26,
          },
          "parametersBytes": 1,
        },
        "FP": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 28,
          },
          "parametersBytes": 1,
        },
        "IN": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 13,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "MF": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "MQ": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              40,
            ],
          },
          "parametersBytes": 4,
        },
        "NF": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              3,
            ],
          },
          "parametersBytes": 4,
        },
        "NP": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              1200,
            ],
          },
          "parametersBytes": 5,
        },
        "NS": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "QS": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 12,
          },
          "parametersBytes": 1,
        },
        "RG": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              -1,
            ],
          },
          "parametersBytes": 8,
        },
        "RI": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "RL": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              100,
            ],
          },
          "parametersBytes": 4,
        },
        "RN": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 11,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "SC": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 14,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "TL": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "TS": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              300,
            ],
          },
          "parametersBytes": 5,
        },
      },
      "readNamesIncluded": true,
      "referenceRequired": false,
      "substitutionMatrix": [
        [
          "C",
          "G",
          "T",
          "N",
        ],
        [
          "A",
          "G",
          "T",
          "N",
        ],
        [
          "A",
          "C",
          "T",
          "N",
        ],
        [
          "A",
          "C",
          "G",
          "N",
        ],
        [
          "A",
          "C",
          "G",
          "T",
        ],
      ],
      "tagEncoding": {},
      "tagIdsDictionary": [
        [],
      ],
    },
    "containerHeader": {
      "_endPosition": 319,
      "_size": 21,
      "alignmentSpan": 300,
      "crc32": 2050599289,
      "landmarks": [
        194,
      ],
      "length": 645,
      "numBases": 500,
      "numBlocks": 13,
      "numLandmarks": 1,
      "numRecords": 5,
      "recordCounter": 0,
      "refSeqId": 0,
      "refSeqStart": 1000,
    },
    "data": [
      {
        "_endPosition": 319,
        "_size": 21,
        "alignmentSpan": 300,
        "crc32": 2050599289,
        "landmarks": [
          194,
        ],
        "length": 645,
        "numBases": 500,
        "numBlocks": 13,
        "numLandmarks": 1,
        "numRecords": 5,
        "recordCounter": 0,
        "refSeqId": 0,
        "refSeqStart": 1000,
      },
      {
        "features": [
          {
            "alignmentStart": 1000,
            "cramFlags": 4,
            "flags": 99,
            "lengthOnRef": 100,
            "mappingQuality": 40,
            "mate": {
              "alignmentStart": 1200,
              "readName": "match",
              "sequenceId": 0,
              "uniqueId": 523,
            },
            "mateRecordNumber": undefined,
            "qualityScores": undefined,
            "readBases": undefined,
            "readFeatures": [
              {
                "code": "B",
                "data": [
                  "R",
                  2,
                ],
                "pos": 1,
                "refPos": 1000,
              },
              {
                "code": "B",
                "data": [
                  "M",
                  55,
                ],
                "pos": 43,
                "refPos": 1042,
              },
              {
                "code": "X",
                "data": 3,
                "pos": 48,
                "refPos": 1047,
              },
              {
                "code": "X",
                "data": 1,
                "pos": 51,
                "refPos": 1050,
              },
              {
                "code": "B",
                "data": [
                  "R",
                  34,
                ],
                "pos": 100,
                "refPos": 1099,
              },
            ],
            "readGroupId": -1,
            "readLength": 100,
            "readName": "match",
            "sequenceId": 0,
            "tags": {},
            "templateLength": 300,
            "templateSize": undefined,
            "uniqueId": 519,
          },
          {
            "alignmentStart": 1100,
            "cramFlags": 3,
            "flags": 0,
            "lengthOnRef": 100,
            "mappingQuality": 40,
            "mate": {
              "alignmentStart": 1200,
              "flags": 0,
              "readName": undefined,
              "sequenceId": 0,
            },
            "qualityScores": [
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
              33,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 100,
            "readName": "qual",
            "sequenceId": 0,
            "tags": {},
            "templateSize": 300,
            "uniqueId": 520,
          },
          {
            "alignmentStart": 1100,
            "cramFlags": 3,
            "flags": 0,
            "lengthOnRef": 100,
            "mappingQuality": 40,
            "mate": {
              "alignmentStart": 1200,
              "flags": 0,
              "readName": undefined,
              "sequenceId": 0,
            },
            "qualityScores": [
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 100,
            "readName": "qstar",
            "sequenceId": 0,
            "tags": {},
            "templateSize": 300,
            "uniqueId": 521,
          },
          {
            "alignmentStart": 1100,
            "cramFlags": 2,
            "flags": 0,
            "lengthOnRef": 100,
            "mappingQuality": 40,
            "mate": {
              "alignmentStart": 1200,
              "flags": 0,
              "readName": undefined,
              "sequenceId": 0,
            },
            "qualityScores": undefined,
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 100,
            "readName": "noqual",
            "sequenceId": 0,
            "tags": {},
            "templateSize": 300,
            "uniqueId": 522,
          },
          {
            "alignmentStart": 1200,
            "cramFlags": 0,
            "flags": 147,
            "lengthOnRef": 100,
            "mappingQuality": 40,
            "mate": {
              "alignmentStart": 1000,
              "readName": "match",
              "sequenceId": 0,
              "uniqueId": 519,
            },
            "qualityScores": undefined,
            "readBases": undefined,
            "readFeatures": [
              {
                "code": "B",
                "data": [
                  "Y",
                  2,
                ],
                "pos": 1,
                "refPos": 1200,
              },
              {
                "code": "B",
                "data": [
                  "Y",
                  2,
                ],
                "pos": 2,
                "refPos": 1201,
              },
              {
                "code": "X",
                "data": 2,
                "pos": 3,
                "refPos": 1202,
              },
              {
                "code": "B",
                "data": [
                  "M",
                  55,
                ],
                "pos": 43,
                "refPos": 1242,
              },
              {
                "code": "X",
                "data": 1,
                "pos": 46,
                "refPos": 1245,
              },
              {
                "code": "X",
                "data": 2,
                "pos": 49,
                "refPos": 1248,
              },
              {
                "code": "X",
                "data": 2,
                "pos": 98,
                "refPos": 1297,
              },
              {
                "code": "B",
                "data": [
                  "Y",
                  34,
                ],
                "pos": 99,
                "refPos": 1298,
              },
              {
                "code": "B",
                "data": [
                  "Y",
                  34,
                ],
                "pos": 100,
                "refPos": 1299,
              },
            ],
            "readGroupId": -1,
            "readLength": 100,
            "readName": "match",
            "sequenceId": 0,
            "tags": {},
            "templateLength": 300,
            "templateSize": undefined,
            "uniqueId": 523,
          },
        ],
        "header": {
          "_endPosition": 562,
          "_size": 44,
          "compressedSize": 40,
          "compressionMethod": "raw",
          "content": Uint8Array [
            0,
            131,
            232,
            129,
            44,
            5,
            0,
            11,
            10,
            11,
            12,
            15,
            16,
            17,
            26,
            27,
            28,
            30,
            31,
            255,
            255,
            255,
            255,
            15,
            188,
            14,
            187,
            152,
            12,
            130,
            56,
            146,
            25,
            54,
            223,
            160,
            201,
            234,
            161,
            96,
          ],
          "contentId": 0,
          "contentPosition": 518,
          "contentType": "MAPPED_SLICE_HEADER",
          "crc32": 1031820642,
          "parsedContent": {
            "_endPosition": 359,
            "_size": 40,
            "contentIds": [
              11,
              12,
              15,
              16,
              17,
              26,
              27,
              28,
              30,
              31,
            ],
            "md5": [
              188,
              14,
              187,
              152,
              12,
              130,
              56,
              146,
              25,
              54,
              223,
              160,
              201,
              234,
              161,
              96,
            ],
            "numBlocks": 11,
            "numContentIds": 10,
            "numRecords": 5,
            "recordCounter": 0,
            "refBaseBlockId": -1,
            "refSeqId": 0,
            "refSeqSpan": 300,
            "refSeqStart": 1000,
          },
          "uncompressedSize": 40,
        },
      },
    ],
  },
]
`;
